Makd.mak

```diff
@@ -565,6 +565,40 @@ $O/test-%.stamp: $O/test-%
 	$(call exec,$< $(ITFLAGS),$<,run)
 	$Vtouch $@
 
+# examples rules
+##########################
+
+# examples are assumed to be standalone programs, so we recursively search
+# for files examples/*.d and assume that each file is a separate example
+# program. Files can exist in sub-directories under examples/ dir.
+# The sources list is filtered through the $(EXAMPLES_FILTER_OUT) variable 
+# contents (using the Make function $(filter-out)), so you can exclude an
+# example by adding the example file path (as an absolute path using $C) to
+# this variable.
+
+# The 'example' target builds all the examples.
+.PHONY: example
+example: $(patsubst $T/example/%.d,$O/example/%,\
+	$(filter-out $(EXAMPLE_FILTER_OUT),$(shell find $T/example/ -name "*.d")))
+
+# General rule to build an example program.
+$O/example/%: BUILD.d.depfile = $O/example/$*.mak
```

Comment:
I wouldn't use a subdirectory because otherwise it needs to be created (look for `setup_build_dir__`). On the other hand, if people want to run the examples manually more often, maybe is better to give them more visibility and put them in a dedicated directory in the build directory instead of the temporary one. We could use `$G/example`.